### PR TITLE
Zgolbj2r: enable users to change their name

### PR DESCRIPTION
**Pull request by AmritSidhu**
Give the ability for users to change there own name
Added new cognito API call update_user_name
Created a new view to change user name
Created a UpdateUserNameEvent
Added controller tests
Added basic system tests

### CODE REVIEW

File: app/controllers/profile_controller.rb [hunk 1]
```diff
@@ -89,4 +115,10 @@ def mfa_page_erros
     @secret_code_svg = generate_new_qr(secret_code: @secret_code, email: current_user.email)
     render :show_change_mfa, status: :bad_request
   end
+
+  def refresh_user(updated_user)
+    current_user.given_name = updated_user.user_attributes.find { |attr| attr.name == 'given_name' }.value
+    current_user.family_name = updated_user.user_attributes.find { |attr| attr.name == 'family_name' }.value
+    current_user.email = updated_user.user_attributes.find { |attr| attr.name == 'email' }.value
```
Comment (jakubmiarka) [inline app/controllers/profile_controller.rb]: we should be updating the current_user.full_name too, given that's what gets displayed. We won't need the `@name = current_user.given_name + " " + current_user.family_name` on line 14 above.

File: config/locales/en.yml [hunk 2]
```diff
@@ -79,6 +79,13 @@ en:
       errors:
         already_exists: "%{email} is already used by a different user."
         generic_error: Error occurred while updating the user's email. Please try again.
+    update_name:
+      title: Change name
+      given_name: Given name
```
Comment (jakubmiarka) [inline config/locales/en.yml]: should this be now First name, as per the latest changes at https://github.com/alphagov/verify-self-service/pull/298 ... you then won't need to use the locales and can rely on the symbol names in the view

File: app/controllers/profile_controller.rb (hunk 1 above)
Comment (jakubmiarka) [reply app/controllers/profile_controller.rb]: actually, we should refactor the full_name construct. It should not be a property on the User model, but rather a method, which takes the current first name and surname. This way we can keep this method as it is (i.e. just updating first name and family name)

Comment (AmritSidhu) [reply app/controllers/profile_controller.rb]: So all instances where `.full_name` is being used, typically on `@user.full_name`, should just become a helper method like `full_name(current_user.first_name, current_user.last_name)` instead?

Comment (jakubmiarka) [reply app/controllers/profile_controller.rb]: nope, `full_name` should not be set in here https://github.com/alphagov/verify-self-service/blob/6f10b297c8b9b249d1c38bb814366c6f5804bbdc/app/models/remote_authenticatable.rb#L43 ... but instead, the User model https://github.com/alphagov/verify-self-service/blob/6f10b297c8b9b249d1c38bb814366c6f5804bbdc/app/models/user.rb should have a method called full_name which returns first_name + last_name, similar to what the TeamMember class does: https://github.com/alphagov/verify-self-service/blob/6f10b297c8b9b249d1c38bb814366c6f5804bbdc/app/models/team_member.rb#L13

File: app/controllers/profile_controller.rb [hunk 3]
```diff
@@ -89,4 +114,10 @@ def mfa_page_erros
     @secret_code_svg = generate_new_qr(secret_code: @secret_code, email: current_user.email)
     render :show_change_mfa, status: :bad_request
   end
+
+  def refresh_user(updated_user)
+    current_user.first_name = updated_user.user_attributes.find { |attr| attr.name == 'given_name' }.value
+    current_user.last_name = updated_user.user_attributes.find { |attr| attr.name == 'family_name' }.value
+    current_user.email = updated_user.user_attributes.find { |attr| attr.name == 'email' }.value
```
Comment (Rosa-Fox) [inline app/controllers/profile_controller.rb]: How come we need to also update the email?

Comment (AmritSidhu) [reply app/controllers/profile_controller.rb]: So it doesn't directly relate to this story. Currently admins have the ability to change a users email, and that email wont show up as changed on the users profile unless they sign out (i think). And we may later chose to add the ability for users themselves to change their email address so therefore is good to have it in place now.

**Pull request merged**